tests/JustSaying.UnitTests/AwsTools/MessageHandling/Sns/TopicByName/WhenSnsTopicTestBase.cs

```diff
@@ -0,0 +1,41 @@
+using System.Threading.Tasks;
+using Amazon.SimpleNotificationService;
+using JustSaying.AwsTools.MessageHandling;
+using NSubstitute;
+using Xunit;
+#pragma warning disable 618
+
+namespace JustSaying.UnitTests.AwsTools.MessageHandling.Sns.TopicByName
+{
+    public abstract class WhenSnsTopicTestBase : IAsyncLifetime
+    {
+        private protected SnsTopicByName SystemUnderTest { get; private set; }
+
+        public IAmazonSimpleNotificationService Sns { get; private set; } = Substitute.For<IAmazonSimpleNotificationService>();
+
+        public virtual async Task InitializeAsync()
+        {
+            Given();
+
+            SystemUnderTest = await CreateSystemUnderTestAsync();
+
+            await WhenAsync().ConfigureAwait(false);
+        }
+
+        public virtual Task DisposeAsync()
+        {
+            if (Sns != null)
```

Comment:
Is this just to satisfy the rule that IDisposable must be disposed? Wonder if we should just ignore the rule as its a substitute.

Reply:
Yeah, I think so, I don't think it hurts but it's a bit noising. I'll push a change with a reduced syntax, there's no need to null these thing out.